pkg/deployment/reconcile/plan_builder_storage.go

```diff
@@ -73,10 +75,33 @@ func createRotateServerStoragePlan(ctx context.Context,
 			}
 
 			if util.StringOrDefault(pvc.Spec.StorageClassName) != storageClassName && storageClassName != "" {
-				// Storageclass has changed
+				// Storage class has changed to non-empty value.
+				if util.StringOrDefault(pvc.Spec.StorageClassName) != "" {
+					// It was set to something else beforehand, so it is not the initialization.
+					pod, ok := cachedStatus.Pod(m.PodName)
+					if !ok {
+						log.Warn().
+							Str("role", group.AsRole()).
+							Str("id", m.ID).
+							Msg("Failed to get pod")
+						continue
+					}
+
+					if _, ok := pod.GetAnnotations()[constants.AnnotationReplaceStorageClassName]; !ok {
```

Comment:
Do not add new annotation:
- in this step just add Condition to member - MemberReplacementRequired

It will be just rotated by current replace flag

Reply:
@ajanikow I need to have a clear description in the ticket of how it should be done. 
For the time being the ticket describes that it should be an annotation: https://arangodb.atlassian.net/browse/TG-181.
Should I add this condition to the ArangoMember resource?

Reply:
changed